yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -616,17 +620,30 @@ private[spark] class Client(
    * throw an appropriate SparkException.
    */
   def run(): Unit = {
-    val (yarnApplicationState, finalApplicationStatus) = monitorApplication(submitApplication())
-    if (yarnApplicationState == YarnApplicationState.FAILED ||
-      finalApplicationStatus == FinalApplicationStatus.FAILED) {
-      throw new SparkException("Application finished with failed status")
-    }
-    if (yarnApplicationState == YarnApplicationState.KILLED ||
-      finalApplicationStatus == FinalApplicationStatus.KILLED) {
-      throw new SparkException("Application is killed")
-    }
-    if (finalApplicationStatus == FinalApplicationStatus.UNDEFINED) {
-      throw new SparkException("The final status of application is undefined")
+    val appId = submitApplication()
+    if (fireAndForget) {
+      logInfo("... waiting before polling ResourceManager for application state")
+      Thread.sleep(5000)
```

Comment:
I would rather see this combined with monitorapplication.  monitorapplication already supports a returnOnRunning option that I think we can use.   There is a waitForApplication in the yarnClientSchedulerBackend.